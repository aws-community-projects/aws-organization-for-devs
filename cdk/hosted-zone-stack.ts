import { Stack, StackProps } from 'aws-cdk-lib';
import { OrganizationPrincipal, Role } from 'aws-cdk-lib/aws-iam';
import { PublicHostedZone } from 'aws-cdk-lib/aws-route53';
import { Construct } from 'constructs';

export class HostedZoneStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const hostedZone = new PublicHostedZone(this, 'RootZone', {
      zoneName: 'awscommunitybuilders.org',
    });

    const role = new Role(this, 'RootZoneOrganizationRole', {
      assumedBy: new OrganizationPrincipal('o-vumqjghgoz'),
      roleName: 'HostedZoneDelegationRole',
    });

    hostedZone.grantDelegation(role);
  }
}
