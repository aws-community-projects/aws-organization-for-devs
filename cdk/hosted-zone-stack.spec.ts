import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { describe, expect, test } from 'vitest';

import { HostedZoneStack } from './hosted-zone-stack';

describe('Entire Stack', () => {
  test('match a snapshot', () => {
    const app = new App();
    const stack = new HostedZoneStack(app, 'test-stack');
    const template = Template.fromStack(stack);

    expect(template).toMatchSnapshot();
  });
});
