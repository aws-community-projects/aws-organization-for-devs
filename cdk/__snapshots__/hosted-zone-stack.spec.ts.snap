// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Entire Stack > match a snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "RootZone831A5F27": {
      "Properties": {
        "Name": "awscommunitybuilders.org.",
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "RootZoneOrganizationRole5B7BD2F8": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Condition": {
                "StringEquals": {
                  "aws:PrincipalOrgID": "o-vumqjghgoz",
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "RoleName": "HostedZoneDelegationRole",
      },
      "Type": "AWS::IAM::Role",
    },
    "RootZoneOrganizationRoleDefaultPolicyD1D7CC29": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "route53:ChangeResourceRecordSets",
              "Condition": {
                "ForAllValues:StringEquals": {
                  "route53:ChangeResourceRecordSetsActions": [
                    "UPSERT",
                    "DELETE",
                  ],
                  "route53:ChangeResourceRecordSetsRecordTypes": [
                    "NS",
                  ],
                },
              },
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":route53:::hostedzone/",
                    {
                      "Ref": "RootZone831A5F27",
                    },
                  ],
                ],
              },
            },
            {
              "Action": "route53:ListHostedZonesByName",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "RootZoneOrganizationRoleDefaultPolicyD1D7CC29",
        "Roles": [
          {
            "Ref": "RootZoneOrganizationRole5B7BD2F8",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
